### ISSUE TITLE
Add option for dosdp query to enable multiple label fields for the generation process.

### ISSUE THREAD

**Issue by matentzn** [INCATools/dosdp-tools#246]
What would be crazy awesome for NLP applications is this:

you have

```
UBERON:001 label "limb" 
exactSynonym "limb2"
relatedSynonym "limb3"
```

Would it be possible to get a setting that allows to generate like from likes?

Example: awesomeAnatomyPattern: `R some X`, NEW:X|UBERON:001

```
NEW:X label: "Awesome Limb"
NEW:X exactSynonym: "Awesome limb" (from label)
NEW:X exactSynonym: "Awesome limb2" (from syn)
NEW:X relatedSynonym: "Awesome limb3"
```

@cmungall double check if I capture cross-product correctly.

**Comment by cmungall** [comment]
Correct, but also the cross-product, e.g `{awesome,brilliant} {limb,limb2,limb3}`

You also need to define a small synonym scope algebra, e.g. ExE=E, ExL=E, RxE=R, BxB=B,Bx{E,R,N}=R,... and also something for types (we need a category for partial abbreviations).

You get a lot of noise, especially with chebi which doesn't respect obo conventions. I think the correct way to deal with this is to treat this as an upstream processing problem (e.g. filter or re-scope syns), and keep the logic and configuration in dosdp-tools simple

This is obviously also useful in LexParse (not yet implemented in dosdp tools, see the original obol approach)

**Comment by matentzn** [comment]
Lets see what Jim thinks, I am not that worried about where this code lives. I would like it to be in dosdp-tools as an optional option, but if I am the only one to do so, lets push it back until a more urgent use case calls for it.

**Comment by meghalithic** [comment]
+1 from me. It'd make it so that I wouldn't have to manually type in synonyms or track them down.

**Comment by aleixpuigb** [comment]
I would love to see this as well. It would be really useful for GWAS catalog. For example, at the moment, users that try searching for GWAS on 'level of protein X in blood', but using the synonym of protein X, can’t find it.

**Comment by matentzn** [comment]
Feature is now in the factory #503

**Comment by balhoff** [comment]
Finally released, thanks @matentzn.
